src/release/main.d

```diff
@@ -238,12 +252,207 @@ string getMilestoneLink ( HTTPConnection con, Repository repo, string ver )
     auto mstone_list = listMilestones(con, repo).find!(a=>a.title == ver);
 
     if (!mstone_list.empty)
+        return mstone_list.front.url ~ "\n\n";
+
+    return "";
+}
+
+
+/*******************************************************************************
+
+    Sends the announcement email
+
+    Params:
+        full_body = complete body of the email, including From & Subject line
+
+*******************************************************************************/
+
+void sendMail ( string full_body )
+{
+    import release.gitHelper;
+
+    import std.process;
+    import std.format;
+
+    auto target = getConfig("neptune.mail-recipient");
+
+    auto proc = pipeShell(format("sendmail %s", target),
+                          Redirect.all);
+
+    proc.stdin.write(full_body);
+    proc.stdin.flush();
+    proc.stdin.close();
+
+    string ret, line;
+
+    wait(proc.pid);
+
+    while ((line = proc.stdout.readln()) !is null)
+        ret ~= line;
```

Comment:
I see trouble here. If the out put is more than 4K, you will be hanging at `wait()` because the kernel will stop the pipe until you read (unless the `pipeShell()` function is reading automatically or something, which it shouldn't).

You should probably read first, and wait after you finished reading. Is stderr being piped too, or it is just printed to this process stderr? If it's piped you should consume it too.